src/main/java/dk/cngroup/trainings/spring/springassignment/config/WebSecurityConfig.java

```diff
@@ -17,23 +18,30 @@
 	protected void configure(HttpSecurity http) throws Exception {
 		http
 				.authorizeRequests()
-				.anyRequest().authenticated()
+				.antMatchers("/web/**")
```

Comment:
so I understand that REST controllers don't have any security now? I think it is fine for the moment, but as one of the next steps you can secure them too (with different kind of authentication than formLogin.

Reply:
I excluded the REST api from authentication, because I have no idea how it should be secured.
Maybe some credentials should be sent in header with every api request?

Reply:
There are several ways of securing REST APIs ranging from simple ones to comples ones :D. Using so called "basic security" should be enough for you (it is basically sending header with username and password in every request), but it is not usually good for production (and if you want to use it anyway, HTTPS is a must!)

Other option is sending some "API key" which basically serves as a username and password.

Other options are more complex - you can for example use a JWT issued by an authentication server or you can use OAUTH2 protocol, where you have a token and you request authentication server to verify it. It is quite complex topic and you can read about it if you want :).

But what you can try (if you want) is login e.g. via Facebook or Twitter. There are guides for it for Spring boot: http://www.baeldung.com/facebook-authentication-with-spring-security-and-social (but I haven't tried this personally, I implemented some JWT solutions, but as I said, you would need an authentication server for that).

Reply:
but maybe implementing login via Facebook will not be so easy for REST, I don't know. I would suggest to go for Basic authentication. One of possible guides: http://www.baeldung.com/spring-security-basic-authentication